deepchem/models/__init__.py

```diff
@@ -148,8 +154,195 @@ def predict(self, dataset, transformers=[]):
+
+    energy_error = y[:,0]-y_pred[:,0]
+    # convert Hartree to kJ/mol
+    energy_error = np.sqrt(np.mean(energy_error*energy_error))*2625.5002
+ 
+    grad = np.reshape(grad, (n_samples, n_atoms, 3))
+    grad_train = np.reshape(grad_train, (n_samples, n_atoms, 3))    
+  
+    grad_error = grad-grad_train
+    # convert Hartree/bohr to kJ/mol/Angstrom
+    grad_error = np.sqrt(np.mean(grad_error*grad_error))*4961.47596096
+
+    print("Energy error (RMSD): %f kJ/mol" % energy_error)
+    print("Grad error (RMSD): %f kJ/mol/A" % grad_error)
+    
+    return energy_error, grad_error
+
+  def evaluate_error_class2(self, dataset, transformers=[]):
+    """
+    Evaluate the error in energy and gradient components, forcebalance-style.
+
+    TODO(rbharath): Should be a subclass PhysicalModel method. Also, need to
+    find a better name for this method (class2 doesn't tell us anything about the
+    semantics of this method.
+    """
+    y_preds = []
+    y_train = []
+    grads = []
+    batch_size = self.model_params["batch_size"]
+    for (X_batch, y_batch, w_batch, ids_batch) in dataset.iterbatches(batch_size):
+
+      # untransformed E is needed for undo_grad_transform
+      energy_batch = self.predict_on_batch(X_batch)
+      grad_batch = self.predict_grad_on_batch(X_batch)
+      grad_batch = undo_grad_transforms(grad_batch, energy_batch, transformers)      
+      grads.append(grad_batch)
+      y_pred_batch = np.reshape(energy_batch, y_batch.shape)
+
+      # y_pred_batch gives us the pred E and pred multitask trained gradE
+      y_pred_batch = undo_transforms(y_pred_batch, transformers)
+      y_preds.append(y_pred_batch)
+
+      # undo transforms on y_batch should know how to handle E and gradE separately
+      y_batch = undo_transforms(y_batch, transformers)
+      y_train.append(y_batch)
+
+    y_pred = np.vstack(y_preds)
+    y = np.vstack(y_train)
+    grad = np.vstack(grads)
+
+    n_samples, n_tasks = len(dataset), len(self.tasks)
+    n_atoms = int((n_tasks-1)/3)
+
+    y_pred = np.reshape(y_pred, (n_samples, n_tasks)) 
+    y = np.reshape(y, (n_samples, n_tasks))
+    grad_train = y[:,1:]
+
+    energy_error = y[:,0]-y_pred[:,0]
+    energy_error = np.sqrt(np.mean(energy_error*energy_error))*2625.5002
+ 
+    grad = np.reshape(grad, (n_samples, n_atoms, 3))
+    grad_train = np.reshape(grad_train, (n_samples, n_atoms, 3))    
+  
+    grad_error = grad-grad_train
+    grad_error = np.sqrt(np.mean(grad_error*grad_error))*4961.47596096
+
+    print("Energy error (RMSD): %f kJ/mol" % energy_error)
+    print("Grad error (RMSD): %f kJ/mol/A" % grad_error)
+    
+    return energy_error, grad_error
+
+  def test_fd_grad(self, dataset, transformers=[]):
+    """
+    Uses self to calculate finite difference gradient on provided Dataset object.
+    Currently only useful if your task is energy and self contains predict_grad_on_batch.
+
+    TODO(rbharath): This shouldn't be a method of the Model class. Perhaps a
```

Comment:
@rbharath This could also be put into `AtomicRegressor` classes if we re-factor hyperparamopt